File: frontend/src/types/appointment.ts
export type AppointmentStatus =
  | "scheduled"
  | "checked_in"
  | "in_progress"
  | "completed"
  | "cancelled"
  | "no_show"

export interface AppointmentPatientRef {
  id: number
  name: string
  medical_record_number: string
  phone_number: string
}

export interface Appointment {
  id: number
  scheduled_at: string
  status: AppointmentStatus
  notes: string | null
  patient: AppointmentPatientRef
  doctor: { id: number; name: string }
  branch: { id: number; name: string }
  created_at: string
}

export interface AppointmentFormData {
  patient_id: number
  doctor_id: number
  branch_id: number
  scheduled_at: string
  notes?: string | null
}

File: mobile/lib/features/schedule/domain/appointment.dart
class Appointment {
  final int id;
  final String scheduledAt;
  final String status;
  final String? notes;
  final String patientName;
  final String patientMrn;
  final String doctorName;

  const Appointment({
    required this.id,
    required this.scheduledAt,
    required this.status,
    required this.notes,
    required this.patientName,
    required this.patientMrn,
    required this.doctorName,
  });

  factory Appointment.fromJson(Map<String, dynamic> json) {
    final patient = json['patient'] as Map<String, dynamic>? ?? const {};
    final doctor = json['doctor'] as Map<String, dynamic>? ?? const {};
    return Appointment(
      id: json['id'] as int,
      scheduledAt: json['scheduled_at'] as String? ?? '',
      status: json['status'] as String? ?? 'scheduled',
      notes: json['notes'] as String?,
      patientName: patient['name'] as String? ?? '-',
      patientMrn: patient['medical_record_number'] as String? ?? '',
      doctorName: doctor['name'] as String? ?? '-',
    );
  }

  String get statusLabel => switch (status) {
        'scheduled' => 'Terjadwal',
        'checked_in' => 'Check-in',
        'in_progress' => 'Diperiksa',
        'completed' => 'Selesai',
        'cancelled' => 'Dibatalkan',
        'no_show' => 'Tidak Hadir',
        _ => status,
      };
}
